corehq/form_processor/management/commands/run_sql.py

```diff
@@ -60,24 +66,42 @@ def confirm(msg):
 
 def timed(func):
     @wraps(func)
-    def timed(sql, dbname):
+    def timed(sql, dbname, *args, **kw):
         print("running on %s database" % dbname)
         start = datetime.now()
         try:
-            return func(sql, dbname)
+            return func(sql, dbname, *args, **kw)
         finally:
             print("{} elapsed: {}".format(dbname, datetime.now() - start))
     return timed
 
 
+def fetch_dicts(cursor):
+    try:
+        rows = cursor.fetchall()
+    except Exception as err:
+        if six.text_type(err) != "no results to fetch":
```

Comment:
It's a generic exception type used for many different db-related errors: `django.db.utils.ProgrammingError: no results to fetch`

I could make the exception a bit more specific, but I didn't think it was worth it since I have to check the message anyway. It's a hack for sure.